README.md

```diff
@@ -12,167 +12,218 @@ Using Kubernetes primitives, administrators configure identities and bindings to
 
 ## Contents
 
-* [Getting Started](#getting-started)
-* [Demo](#demo)
+
 * [Components](#components)
-* [Features](docs/readmes/README.md#features)
-* [What To Do Next?](#what-to-do-next)
+  + [Managed Identity Controller](#managed-identity-controller)
+  + [Node Managed Identity](#node-managed-identity)
+* [Demo](#demo)
+  + [1. Deploy aad-pod-identity](#1-deploy-aad-pod-identity)
+  + [2. Create an identity on Azure](#2-create-an-identity-on-azure)
+  + [3. Deploy AzureIdentity](#3-deploy-azureidentity)
+  + [4. (Optional) Match pods in the namespace](#4--optional--match-pods-in-the-namespace)
+  + [5. Deploy AzureIdentityBinding](#5-deploy-azureidentitybinding)
+  + [6. Deployment and Validation](#6-deployment-and-validation)
+* [Uninstall Notes](#uninstall-notes)
+* [What To Do Next?](#what-to-do-next-)
 * [Code of Conduct](#code-of-conduct)
 
 ## Getting Started
 
-### Prerequisites
+It is recommended to get familiar with the AAD Pod Identity ecosystem before diving into the demo. It consists of the Managed Identity Controller (MIC) deployment, the Node Managed Identity (NMI) DaemonSet, and several standard and custom resources.
+
+## Components
+
+AAD Pod Identity has two components: the [Managed Identity Controller] (MIC) and the [Node Managed Identity] (NMI).
+
+### Managed Identity Controller
+
+The Managed Identity Controller (MIC) is a Kubernetes [custom resource] that watches for changes to pods, identities, and bindings through the Kubernetes API server. When it detects a relevant change, the MIC adds or deletes assigned identities as needed.
+
+Specifically, when a pod is scheduled, the MIC assigns an identity to the underlying VM during the creation phase. When the pod is deleted, it removes the assigned identity from the VM. The MIC takes similar actions when identities or bindings are created or deleted.
+
+### Node Managed Identity
+
+The authorization request to fetch a Service Principal Token from an MSI endpoint is sent to a standard Instance Metadata endpoint which is redirected to the NMI pod. The redirection is accomplished by adding rules to redirect POD CIDR traffic with metadata endpoint IP on port 80 to the NMI endpoint. The NMI server identifies the pod based on the remote address of the request and then queries Kubernetes (through MIC) for a matching Azure identity. NMI then makes an Azure Active Directory Authentication Library ([ADAL]) request to get the token for the client id and returns it as a response. If the request had client id as part of the query, it is validated against the admin-configured client id.
+
+Here is an example cURL command that will fetch an Azure KeyVault token from within a pod identified by an AAD-Pod-Identity selector:
+
+```bash
+curl http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://vault.azure.net
```

Comment:
can we replace the curl command with the example from here - https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-curl and also add the link to the doc if users want to refer?